### ISSUE TITLE
New-AzureRmResourceGroupDeployment fail when the first interval check

### ISSUE THREAD

**Issue by LiliDeng** [Azure/azure-powershell#2416]
Run below command, it will fail in a high rate when the first 5 seconds interval check, do we have any workaround to increase the interval or any other parameter can be used to avoid this issue?
Command:
New-AzureRmResourceGroupDeployment -Name $ResourceGroupDeploymentName -ResourceGroupName $RGName -TemplateFile $TemplateFile -TemplateParameterFile $TemplateParameterFile -Verbose
Failure:
VERBOSE: 7:16:38 PM - Checking deployment status in 5 seconds
New-AzureRmResourceGroupDeployment : Deployment 'ICA-RG-A1-liliarmde-6-6-19-16-22-deployment' could not be found.

**Comment by markcowl** [comment]
@LiliDeng Can you try executing with ```$DebugPreference="Continue"``` to get debugging information for this issue.

@ravbhatnagar can you take a look?

**Comment by ZACommando** [comment]
My team and I are experiencing the same issue.  We are running the latest version of the Azure PowerShell (1.5 released June 1st) and the three of us keep getting the failure at the same point: Checking deployment status in 5 seconds.

The deployments are completing successfully, but the remainder of our deployment script is dependent upon the New-AzureRmResourceGroupDeployment cmdlet succeeding.

Here is the cmdlet call:
```
New-AzureRmResourceGroupDeployment 
  -Name azuredeploy-0610-2206 
  -ResourceGroupName [ResourceGroupName] 
  -TemplateFile C:\azuredeploy.json
  -TemplateParameterFile C:\azuredeploy.parameters.json
  -Force
  -Verbose
  -Mode Incremental
```

Here is the output from the cmdlet with `$DebugPreference="Continue"` (shortened for brevity and security):
```
VERBOSE: 5:07:07 PM - Template is valid.
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:07 PM: bbacb506-eba4-4988-ad4a-e959f3723eed - AcquireTokenHandlerBase: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/f2e3e4c3-e58f-4b8a-bde8-550f8708312a/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache (6 items)
	Authentication Target: User
	

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 6/10/2016 10:07:07 PM: bbacb506-eba4-4988-ad4a-e959f3723eed - TokenCache: Looking up cache for a token...

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:07 PM: bbacb506-eba4-4988-ad4a-e959f3723eed - TokenCache: An item matching the requested resource was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 6/10/2016 10:07:07 PM: bbacb506-eba4-4988-ad4a-e959f3723eed - TokenCache: 23.916049705 minutes left until token in cache expires

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:07 PM: bbacb506-eba4-4988-ad4a-e959f3723eed - TokenCache: A matching item (access token or refresh token or both) was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:07 PM: bbacb506-eba4-4988-ad4a-e959f3723eed - AcquireTokenHandlerBase: === Token Acquisition finished successfully. An access token was retuned:
	Access Token Hash: R0/KcY2C3oRCzNjySDx5UntQjnu/m/a9Qno+NwIVAIg=
	Refresh Token Hash: PfqkqzOnkXiWdFcM+WiHtopYSxbDzQSAeiXPbHWOrTQ=
	Expiration Time: 6/10/2016 10:31:02 PM +00:00
	User Hash: VRIHvXRrJ1eMOuP8Yc5z9PcrbiRXYotxOpR1VVD0GF8=
	

DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/[subscriptionID]/resourcegroups/[ResourceGroupName]/providers/Microsoft.Resources/deployments/azuredeploy-0610-2206?a
pi-version=2016-02-01

Headers:
x-ms-client-request-id        : 714582a7-7b01-4316-8e94-8bb40174b88b
accept-language               : en-US

Body:
{
  "properties": {
    "template": [valid template],
    "parameters": [valid parameters],
    "mode": "Incremental"
  }
}


VERBOSE: 5:07:07 PM - Create template deployment 'azuredeploy-0610-2206'
VERBOSE: 5:07:07 PM - Checking deployment status in 5 seconds
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 43d926cb-48e4-4e83-947b-5c946a7cc19d - AcquireTokenHandlerBase: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/f2e3e4c3-e58f-4b8a-bde8-550f8708312a/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache (6 items)
	Authentication Target: User
	

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 6/10/2016 10:07:12 PM: 43d926cb-48e4-4e83-947b-5c946a7cc19d - TokenCache: Looking up cache for a token...

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 43d926cb-48e4-4e83-947b-5c946a7cc19d - TokenCache: An item matching the requested resource was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 6/10/2016 10:07:12 PM: 43d926cb-48e4-4e83-947b-5c946a7cc19d - TokenCache: 23.83216428 minutes left until token in cache expires

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 43d926cb-48e4-4e83-947b-5c946a7cc19d - TokenCache: A matching item (access token or refresh token or both) was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 43d926cb-48e4-4e83-947b-5c946a7cc19d - AcquireTokenHandlerBase: === Token Acquisition finished successfully. An access token was retuned:
	Access Token Hash: R0/KcY2C3oRCzNjySDx5UntQjnu/m/a9Qno+NwIVAIg=
	Refresh Token Hash: PfqkqzOnkXiWdFcM+WiHtopYSxbDzQSAeiXPbHWOrTQ=
	Expiration Time: 6/10/2016 10:31:02 PM +00:00
	User Hash: VRIHvXRrJ1eMOuP8Yc5z9PcrbiRXYotxOpR1VVD0GF8=
	

DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/[subscriptionID]/resourcegroups/[ResourceGroupName]/providers/Microsoft.Resources/deployments/azuredeploy-0610-2154/o
perationStatuses/08587360112064002978?api-version=2016-02-01

Headers:

Body:



DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 9ef86634-3120-4b8f-b591-ec29e32712ca - AcquireTokenHandlerBase: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/f2e3e4c3-e58f-4b8a-bde8-550f8708312a/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache (6 items)
	Authentication Target: User
	

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 6/10/2016 10:07:12 PM: 9ef86634-3120-4b8f-b591-ec29e32712ca - TokenCache: Looking up cache for a token...

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 9ef86634-3120-4b8f-b591-ec29e32712ca - TokenCache: An item matching the requested resource was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 6/10/2016 10:07:12 PM: 9ef86634-3120-4b8f-b591-ec29e32712ca - TokenCache: 23.8315320483333 minutes left until token in cache expires

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 9ef86634-3120-4b8f-b591-ec29e32712ca - TokenCache: A matching item (access token or refresh token or both) was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 6/10/2016 10:07:12 PM: 9ef86634-3120-4b8f-b591-ec29e32712ca - AcquireTokenHandlerBase: === Token Acquisition finished successfully. An access token was retuned:
	Access Token Hash: R0/KcY2C3oRCzNjySDx5UntQjnu/m/a9Qno+NwIVAIg=
	Refresh Token Hash: PfqkqzOnkXiWdFcM+WiHtopYSxbDzQSAeiXPbHWOrTQ=
	Expiration Time: 6/10/2016 10:31:02 PM +00:00
	User Hash: VRIHvXRrJ1eMOuP8Yc5z9PcrbiRXYotxOpR1VVD0GF8=
	

DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/[subscriptionID]/resourcegroups/[ResourceGroupName]/deployments/azuredeploy-0610-2206/operations?api-version=2016-02-
01

Headers:
x-ms-client-request-id        : 84493dc9-a3f3-4f33-add0-b7434fb908a4
accept-language               : en-US

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Pragma                        : no-cache
x-ms-ratelimit-remaining-subscription-reads: 14954
x-ms-request-id               : 6c4ddb77-17f9-4996-814b-819024ef0635
x-ms-correlation-request-id   : 6c4ddb77-17f9-4996-814b-819024ef0635
x-ms-routing-request-id       : NORTHCENTRALUS:20160610T220712Z:6c4ddb77-17f9-4996-814b-819024ef0635
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Cache-Control                 : no-cache
Date                          : Fri, 10 Jun 2016 22:07:12 GMT

Body:
{
  "status": "Running"
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
NotFound

Headers:
Pragma                        : no-cache
x-ms-failure-cause            : gateway
x-ms-ratelimit-remaining-subscription-reads: 14994
x-ms-request-id               : 85ce6d9c-d674-43cb-848d-4c338ca7b27e
x-ms-correlation-request-id   : 85ce6d9c-d674-43cb-848d-4c338ca7b27e
x-ms-routing-request-id       : NORTHCENTRALUS:20160610T220712Z:85ce6d9c-d674-43cb-848d-4c338ca7b27e
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Cache-Control                 : no-cache
Date                          : Fri, 10 Jun 2016 22:07:12 GMT

Body:
{
  "error": {
    "code": "DeploymentNotFound",
    "message": "Deployment 'azuredeploy-0610-2206' could not be found."
  }
}


New-AzureRmResourceGroupDeployment : Deployment 'azuredeploy-0610-2206' could not be found.
At C:\Deploy-AzureResourceGroup.ps1:157 char:11
+ $output = New-AzureRmResourceGroupDeployment -Name ((Get-ChildItem $T ...
+           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzureRmResourceGroupDeployment], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDeploymentCmdlet
 
DEBUG: 5:07:12 PM - NewAzureResourceGroupDeploymentCmdlet end processing.
DEBUG: 5:07:12 PM - NewAzureResourceGroupDeploymentCmdlet end processing.
```

**Comment by TianoMS** [comment]
I'll look into issue.

**Comment by brianmowens** [comment]
I'm experience the same issue as ZACommando, except my script calls a new deployment twice, and uses different ARM templates for each deployment. One of these templates reliably fails each time it's deployed with a not found error message, and the second template never fails. Is there anything I can attempt providing to help debug this?

Also on version 1.5.0.

**Comment by ravbhatnagar** [comment]
@brianmowens  @ZACommando  @LiliDeng We understand why this issue is happening. We will address this in the release that goes out in the first week of July. Would it be possible for you to share the templates? It will help in verifying the fix. Thanks!

**Comment by ravbhatnagar** [comment]
Fixed in 1.5.1
